scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,54 +58,48 @@ def generate_load_table_files(dataset_tables_dict):
+            print(dataset.keys())
+            exit()
 
-        # TODO: edit the value = dataframe for a table to have the entity:table_name_id format
+        errors = validate_df(field_dict=field_dict, schema_column_list=schema_dict[table], data_df=dataset[table])
+        # get indices of the failed rows
+        errors_index_rows = [e.row for e in errors]
 
-        # add KVP {table name : table_df} entry to all tables dictionary
-        dataset_tables_dict[table] = table_df
+        # if any errors, print error message and exit
+        if errors_index_rows:
+            print(f"Failed: Dataset validation errors found in table {table}. Please retry after correcting errors listed in validation_errors.csv.")
+            pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
+            exit()
 
-    print(f"Success: Dataset has been partitioned into {len(dataset_table_names)} tables: {dataset_table_names}")
-    return dataset_tables_dict
+        # TODO: determine how to handle drops of failed rows/columns to get cleaned data
+        validated_dataset[table] = dataset[table]
+        print(f"Success: Table {table} has been validated.")
+    
+    print(f"Success: Dataset has been validated.")
+    return validated_dataset
 
 
-def validate_dataset(dataset):
-    """Validate the full dataset metadata with Schema validator."""
+def col_is_in_defined_fields(col_name, field_dict):
+    list_of_fields = list(field_dict.keys())
+    if col_name in list_of_fields:
+        return True
 
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
+    return False
 
-    # if any errors, print error message and exit
-    if errors_index_rows:
-        print(f"Failed: Dataset validation errors found. Please retry after correcting errors listed in validation_errors.csv.")
-        pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
-        exit()
 
-    # TODO: determine how to handle drops of failed rows/columns to get cleaned data
-    # errors_index_rows = [e.row for e in errors]
-    # data_clean = dataset_metadata_df.drop(index=errors_index_rows, axis=1)
-    # data_clean.to_csv('clean_data.csv')
-    print(f"Success: Dataset has been validated.")
-    return dataset
-
-
-def load_excel_input(excel, expected_dataset_cols):
+def load_excel_input(excel, allowed_dataset_cols):
     """Read excel file input into a dataframe and validate for required columns."""
 
     try:
-        dataset_metadata_df = pd.read_excel(excel, sheet_name="Sheet1", skiprows=2, usecols=expected_dataset_cols, index_col=None)
-
+        raw_dataset_df = pd.read_excel(excel, sheet_name=None, skiprows=2, index_col=None, usecols=lambda x: x in allowed_dataset_cols)
```

Comment:
why skiprows=2? is there a description somewhere of the expected formatting of the excel input?

Reply:
also it looks like usecols can be a list, so I don't think you need a lambda here - you can just have `usecols=allowed_dataset_cols`. https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html

Reply:
If usecols is a list, it requires that ALL the cols be present. That's not what I want. I want it to use it as a filter, not a strict requirement. That's why I changed it. 

skiprows=2 ... as far as I can tell, that's the expected input format. I'm not sure if it's described anywhere, but it is displayed in the example data I provide in the README. 

I can ask Sam to provide a better description of expected input format.

Reply:
Required columns is covered in validation code, not ingest code. I separated it out because I don't think it is good practice to mix together ingest and validation, especially not here. 

It should skip the first two rows - those are headers. The data starts one row 3.

Reply:
for usecols: got it, thanks, so basically here as you're reading in the data you're filtering down the column names to the possible expected columns, i.e. dropping any data that's not in your expected list of fields; later you'll check for required ones. makes sense. I'd add a comment to explain the use of a lambda here rather than a list - that's cool functionality but not obvious to the naive reader.

for skiprows: awesome, that's what I was trying to get at with the expected formatting of the excel input. a comment to the effect of "skip 2 rows because the data file has 2 header rows" would be great here.

Reply:
Done!